Couldn't generate from external-secrets-operator crds. Provider collision.

### What happened?

External secrets is having conflicting property `provider`. Maybe the property should be mapped to _provider or $provider.



### Example

Running
```bash
helm template --repo https://charts.external-secrets.io ex external-secrets --version 0.19.2 --include-crds | crd2pulumi --nodejsPath ./test - --force
```
got
```log
panic: fatal: error An assertion has failed: could not parse Pulumi package. source error: could not import spec: error binding resource kubernetes:external-secrets.io/v1:ClusterSecretStore: failed to bind properties for kubernetes:external-secrets.io/v1:ClusterSecretStore: error binding type for property "spec": error binding type for property "provider": invalid property names

goroutine 1 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        github.com/pulumi/pulumi/sdk/v3@v3.188.0/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.AssertNoErrorf({0x103719448, 0x140033d61c0}, {0x102bcd430?, 0x14001bbf020?}, {0x0?, 0x14001bbf001?, 0xb?})
        github.com/pulumi/pulumi/sdk/v3@v3.188.0/go/common/util/contract/assert.go:51 +0xfc
github.com/pulumi/crd2pulumi/pkg/codegen.(*PackageGenerator).SchemaPackageWithObjectMetaType(0x1400035a690)
        github.com/pulumi/crd2pulumi/pkg/codegen/packagegenerator.go:121 +0x64
github.com/pulumi/crd2pulumi/pkg/codegen.GenerateNodeJS(0x102b94628?, 0x14000a9d020)
        github.com/pulumi/crd2pulumi/pkg/codegen/nodejs.go:33 +0x28
github.com/pulumi/crd2pulumi/pkg/codegen.Generate(0x14000a9d020, {0x1400089fbd8, 0x1, 0x1})
        github.com/pulumi/crd2pulumi/pkg/codegen/codegen.go:62 +0x188
github.com/pulumi/crd2pulumi/cmd.New.func3(0x14000a45000?, {0x14000a0fec0, 0x1, 0x4})
        github.com/pulumi/crd2pulumi/cmd/root.go:95 +0x30c
github.com/spf13/cobra.(*Command).execute(0x14000a7c608, {0x140001e6010, 0x4, 0x4})
        github.com/spf13/cobra@v1.9.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000a7c608)
        github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(0x0?)
        github.com/spf13/cobra@v1.9.1/command.go:1071 +0x1c
main.main()
        github.com/pulumi/crd2pulumi/main.go:25 +0x20
```

### Output of `pulumi about`

CLI          
Version      3.204.0
Go Version   go1.25.3
Go Compiler  gc

Host     
OS       darwin
Version  14.5
Arch     arm64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/Yukimotochern
User           Yukimotochern
Organizations  Yukimotochern
Token type     personal

Pulumi locates its logs in /var/folders/2f/zsz2hxsx4hj3tn828px58fv40000gp/T/ by default
warning: Failed to read project: no Pulumi.yaml project file found (searching upwards from /Users/chen/projects/nearme-infra). If you have not created a project yet, use `pulumi new` to do so: no project file found
warning: Failed to get information about the current stack: no Pulumi.yaml project file found (searching upwards from /Users/chen/projects/nearme-infra). If you have not created a project yet, use `pulumi new` to do so: no project file found
warning: A new version of Pulumi is available. To upgrade from version '3.204.0' to '3.210.0', run 
   $ curl -sSL https://get.pulumi.com | sh
or visit https://pulumi.com/docs/install/ for manual instructions and release notes.

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
 	github.com/go-openapi/jsonreference v0.21.0
 	github.com/iancoleman/strcase v0.3.0
 	github.com/pulumi/pulumi-java/pkg v1.16.1
-	github.com/pulumi/pulumi-kubernetes/provider/v4 v4.0.0-20260103103121-fd2b31702d9d
+	github.com/pulumi/pulumi-kubernetes/provider/v4 v4.0.0-20260115033456-f47bc2d6f199
 	github.com/pulumi/pulumi/pkg/v3 v3.215.0
 	github.com/pulumi/pulumi/sdk/v3 v3.215.0
 	github.com/spf13/cobra v1.10.1

